tests/cursor/bug1839-001.phpt

```diff
@@ -0,0 +1,54 @@
+--TEST--
+PHPC-1839: Referenced, out-of-scope, non-interned string in typeMap
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+function createTypemap()
```

Comment:
Using a function here to reproduce the behaviour of php-di. Notably this produces a different zval than in test 002 (see below).